Add-page

thampaponn opened the pull request:
input type file เอาไงดี

thampaponn committed:
SafeAreaView & Fix bugs

added SafeAreaView for styling, fix key bug

thampaponn committed:
Merge branch 'main' into addpage

thampaponn committed:
Update Add port screen

thampaponn committed:
backend

thampaponn committed:
Update add port

thampaponn committed:
Fix InputForm component

use Input from rneui in InputForm component for reusing

madara1150 commented on screens/AddPortScreen.tsx:
```diff
@@ -1,10 +1,23 @@
-    </View>
+    <SafeAreaView style={{ backgroundColor: "#FFFFFF" }}>
+      <Card containerStyle={{ borderRadius: 20, padding: 25 }}>
+      <InputForm title='ชื่อโครงงานภาษาไทย' />
```
อย่าใช้ inputForm ให้เรียกใช้ Input ตรงๆเลย เพราะจะทำ backend ยาก
```suggestion
      <Text style={styles.formText}>{'ชื่อโครงงานภาษาไทย'}</Text>
      <Input  style={styles.textInput}  inputContainerStyle={{ borderBottomWidth: 0 }}/>
```

madara1150 commented on components/ui/InputForm.tsx:
ลบไฟล์นี้ออกเลยก็ได้ห้ามใช้

madara1150 commented on screens/AddPortScreen.tsx:
```diff
@@ -1,10 +1,23 @@
-import { View, Text } from 'react-native'
+import { View, Text, SafeAreaView } from 'react-native'
+import { Button, Card, Chip, Input } from "@rneui/themed";
```
อันไหนไม่ใช้ลบออก

madara1150 commented on screens/AddPortScreen.tsx:
```diff
@@ -1,10 +1,23 @@
-import { View, Text } from 'react-native'
+import { View, Text, SafeAreaView } from 'react-native'
```
อันไหนไม่ใช้ลบออก

madara1150 commented on screens/AddPortScreen.tsx:
```diff
@@ -1,10 +1,23 @@
+import { View, Text, SafeAreaView } from 'react-native'
+import { Button, Card, Chip, Input } from "@rneui/themed";
 import React from 'react'
+import { InputForm } from '../components/ui/InputForm';
```
ลบออก

madara1150 commented on screens/MainPageScreen.tsx:
```diff
@@ -55,8 +55,9 @@ const MainPageScreen = () => {
       <ScrollView>
         <ProudctFilter type={categoryMockup} />
-        {PostMockup.map((post) => (
+        {PostMockup.map((post, key) => (
           <ProductCard
+            key={key}
```
ควรเป็น index
```suggestion
        {PostMockup.map((post, index) => (
          <ProductCard
            key={index}
```

madara1150 commented on screens/NotificationScreen.tsx:
```diff
@@ -48,8 +48,8 @@ export default function NotificationScreen() {
           {mockupData &&
-            mockupData.map((data) => (
-              <View style={{ marginTop: 10 }}>
+            mockupData.map((data, key) => (
+              <View key={key} style={{ marginTop: 10 }}>
```
```suggestion
            mockupData.map((data, index) => (
              <View key={index} style={{ marginTop: 10 }}>
```

madara1150 commented on screens/ProductDetail.tsx:
```diff
@@ -69,8 +69,9 @@ const ProductDetail = () => {
             รูปที่เกี่ยวข้อง
           </Text>
-          {ImageMockup.map((image) => (
+          {ImageMockup.map((image, key) => (
             <Card.Image
+              key={key}
```
```suggestion
          {ImageMockup.map((image, index) => (
            <Card.Image
              key={index}
```

madara1150 commented on screens/ProductDetail.tsx:
```diff
@@ -82,8 +83,8 @@ const ProductDetail = () => {
           </Text>
-          {fileMockup.map((file) => (
-            <Card containerStyle={{ borderRadius: 12 }}>
+          {fileMockup.map((file, key) => (
+            <Card key={key} containerStyle={{ borderRadius: 12 }}>
```
```suggestion
          {fileMockup.map((file, index) => (
            <Card key={index} containerStyle={{ borderRadius: 12 }}>
```

madara1150 commented:
@thampaponn  แก้เสดแล้ว แคปรูปลงไว้ด้วย

thampaponn committed:
CSS fixed

fix margin top, add port page fields added

thampaponn committed:
Refactored

Delete InputForm component(unused)
Delete unused import
change from key to index

thampaponn committed:
Remove InputForm

ฝากทำให้เป็น TextInput ที ในหน้า SignUp, StudentSignUp, ProfessorSignUp, VisitorSignUp

thampaponn commented:
AddPortScreen
<img width="1265" alt="image" src="https://github.com/thampaponn/GoodPort/assets/88426179/075f2483-71d8-492d-8fd3-d5235dc595dc">

thampaponn commented:
MainPageScreen
<img width="584" alt="image" src="https://github.com/thampaponn/GoodPort/assets/88426179/d2d07d76-3a5c-4afd-b4f1-0057c9e7c434">

thampaponn commented:
NotificationScreen
<img width="739" alt="image" src="https://github.com/thampaponn/GoodPort/assets/88426179/2b6eae58-7ee4-489b-9f44-201c9c966c5f">
<img width="739" alt="image" src="https://github.com/thampaponn/GoodPort/assets/88426179/82e8e5cb-c520-4dab-9c14-59924e4dbfba">

thampaponn commented:
ProductDetail
<img width="739" alt="image" src="https://github.com/thampaponn/GoodPort/assets/88426179/5cf34cd0-7794-4521-a70d-4a5cdba6d6c5">

thampaponn committed:
Update NotificationScreen.tsx

madara1150 commented:
@thampaponn merge ได้
